src/ts/layers/choropleth.ts

```diff
@@ -99,43 +115,37 @@ module powerbi.extensibility.visual {
                 let filter = ['in', choroSettings.vectorProperty];
                 const choroplethData = this.source.getData(map, settings);
 
-                let existingStops = {};
-                let validStops = true;
+                let existingStops = {};                
 
                 for (let row of choroplethData) {
                     const location = row[roleMap.location.displayName];
 
-                    if (location == null || location == undefined) {
+                    let color: any = getColorStop(row[roleMap.color.displayName]);
+                    let outlineColor: any = getColorStop(row[roleMap.color.displayName]);
+
+                    if (!location || !color || !outlineColor) {
                         // Stop value cannot be undefined or null; don't add this row to the stops
                         continue;
                     }
-
-                    let color: any = colorStops(row[roleMap.color.displayName]);
-                    let outlineColor: any = colorStops(row[roleMap.color.displayName]);
+                
                     outlineColor = outlineColor.darken(2);
 
                     if (existingStops[location]) {
                         // Duplicate stop found. In case there are many rows, Mapbox generates so many errors on the
                         // console, that it can make the entire Power BI plugin unresponsive. This is why we validate
-                        // the stops here, and won't let invalid stops to be passed to Mapbox.
-                        validStops = false;
-                        break;
+                        // the stops here, and won't let invalid stops to be passed to Mapbox.                        
+                        continue;
```

Comment:
I really don't think this is an equivalent change. Choropleth visualizations can be configured in such ways (for example by not summarizing the color property) that will result in multiple color stops for one single location. And by this `continue` instead of `break`, you seem to pick the first stop for the location and visualize that. And this can lead to incorrect and misleading color values overall.

Reply:
@ryanbaumann , I just discussed about this code change with @szilardhuber , and I really don't think that this change would be a wise decision. After this change we would most probably end up with **incorrect colour values**.

So this change would enable us to show colour values for US counties, but the problem is that for example there are like 21 counties called Washington in the entire US. That would result in 21 color stops for Washington, and multiple stops for one single value is an error. This code actually picks the first stop, and will map the same colour for all the 21 Washington counties. Which I believe is really incorrect. In this case we would visualise that the exact same amount of people voted on Republicans in all Washington counties for example, which is obviously not true.

I also believe that the source of the issue is that we are using `name` as vector property for US counties too, but in case of US counties name does not identify one single county. I think we should consider having `admin_code` (or some sort of FIPS code of the county) as vector property.

Was it clear? What do you think? Any ideas?

Reply:
I agree @baspeti - let's have @kovacsgz explain their reasoning for the PR, and we can review afterward.  

Good call on the unique property value for each vector property - I agree.  Please open a ticket for that and I can update those vector tilesets to use a unique feature id.